### ISSUE TITLE
Current Network doesn't work at all with classifier

### ISSUE THREAD

**Issue by PutawanDE** [PutawanDE/MLP-261456#1]
When training CrossPat (Assignment task 2), the neural network doesn't learn. I tried replicating this issue using [ConvnetJS online neural network visualization](https://cs.stanford.edu/people/karpathy/convnetjs/demo/classify2d.html). The neural network acted as expected, I continued trying to fix this issue using this tool. I found that the below configuration worked. 
```js
layer_defs = [];
layer_defs.push({type:'input', out_sx:1, out_sy:1, out_depth:2});
layer_defs.push({type:'fc', num_neurons:4, activation: 'tanh'});
// layer_defs.push({type:'fc', num_neurons:2, activation: 'sigmoid'});
layer_defs.push({type:'softmax', num_classes:2});

net = new convnetjs.Net();
net.makeLayers(layer_defs);

trainer = new convnetjs.SGDTrainer(net, {learning_rate:0.01, momentum:0.1, batch_size:10, l2_decay:0.001});
```
The output layer activation function is softmax. Tanh made the network converge really fast. Further parameter adjustment is needed, though this works a lot better than the current training configuration. Digging deeper into the working of this simulation, I found it uses the class negative log likelihood function, instead of my MSE loss function. Many sources also suggest using a different loss function and output layer activation function for classifiers.

Sources:
https://www.pinecone.io/learn/softmax-activation/
https://www.researchgate.net/post/What_does_the_Activation_Function_at_Output_layer_do
https://python-course.eu/machine-learning/softmax-as-activation-function.php
https://www.youtube.com/watch?v=8ps_JEW42xs
Code:
https://github.com/karpathy/convnetjs/blob/master/src/convnet_layers_loss.js

According to the sources, for this task, only one output node is enough. This will become a binary classifier. The sigmoid function can be used in the output layer as it is equivalent to the softmax function. 

What to do

- [x] Implement output layer activation function as well as the derivative
- [x] Specify the type of network to be able to choose between 2 different loss functions
~~- [ ] Implement argmax to be able to expand the output layer ?~~

**Comment by PutawanDE** [comment]
Fix with commit a1dd56e7cf4cb6e4b1e2be619319f4a17a8ad94b, a1dd56e7cf4cb6e4b1e2be619319f4a17a8ad94b. All Done.
